### PR TITLE
Fixes for AbstractConjunctions bugs

### PR DESCRIPTION
**Pull request by burkartM**
Fixes for error and bug mentioned in #87

### CODE REVIEW

File: ConfigSpace/configuration_space.py [hunk 1]
```diff
@@ -233,7 +233,7 @@ def add_conditions(self, conditions: List[ConditionComponent]) -> List[Condition
                 edges.extend(
                     [(dlc.parent, dlc.child) for dlc in dlcs])
                 values.extend([dlc.value for dlc in dlcs])
-                conditions_to_add.extend(dlcs)
+                conditions_to_add.extend([condition] * len(dlcs))
```
Comment (mfeurer) [inline ConfigSpace/configuration_space.py]: Could you please explain this statement?

Comment (mfeurer) [reply ConfigSpace/configuration_space.py]: Okay, this seems to fix your statement about adding Conjunctions. Could you please provide a failing example for this?

**Comment by burkartM** [comment]
I added a unit test  that should fail on the "old" version.
Explanation for:
https://github.com/automl/ConfigSpace/blob/58321ea76e3c34043a9250ba5ccda1f880447abc/ConfigSpace/configuration_space.py#L236
I compared add_condition (which works) with add_condtions.
The only difference I could find is that while in add_condition you add for every the actual condition as condition:
https://github.com/automl/ConfigSpace/blob/58321ea76e3c34043a9250ba5ccda1f880447abc/ConfigSpace/configuration_space.py#L203-L207
you add the dlc as condition in add_conditions:
https://github.com/automl/ConfigSpace/blob/58321ea76e3c34043a9250ba5ccda1f880447abc/ConfigSpace/configuration_space.py#L236
https://github.com/automl/ConfigSpace/blob/58321ea76e3c34043a9250ba5ccda1f880447abc/ConfigSpace/configuration_space.py#L241-L242
This is why [condition] * len(dlcs) should fix the issue, because the zip will now map each edge to the actual condition and not the according dlc.

**Comment by mfeurer** [comment]
Thanks a lot!

**Pull request merged**